core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -164,35 +164,26 @@ sealed trait Chunk[+A] extends ChunkLike[A] { self =>
   }
 
   def dropWhileM[R, E](p: A => ZIO[R, E, Boolean]): ZIO[R, E, Chunk[A]] = {
-    val len                                       = self.length
-    var dest: ZIO[R, E, (Boolean, Int, Array[A])] = UIO.succeedNow((true, 0, null.asInstanceOf[Array[A]]))
+    val len     = self.length
+    val builder = ChunkBuilder.make[A]()
+    builder.sizeHint(len)
+    var dest: ZIO[R, E, (Boolean, ChunkBuilder[A])] = IO.succeedNow((true, builder))
```

Comment:
There is no need to pass the `builder` into the fold, because the builder is a constant outside the loop and can be referred to in the functions below.